lib/routes/wuthering-waves/news.ts

```diff
@@ -0,0 +1,167 @@
+
+    /** The amount of articles to fetch. */
+    const limit = Math.max(limitParsed, 0);
+
+    /** The language to fetch articles in. */
+    const language = ctx.req.param(Parameters.Language) ?? 'en';
+
+    /** The category ID to filter articles on. */
+    const categoryIdParameter = ctx.req.param(Parameters.Category);
+    const categoryId = categoryIdParameter ? Number.parseInt(categoryIdParameter) : null;
+
+    if (!SUPPORTED_LANGUAGES.includes(language)) {
+        throw new Error(`"${language}" is not a supported language. Please choose between the following: ${SUPPORTED_LANGUAGES.join(', ')}.`);
+    }
+
+    const supportedCategoryIds = Object.values(ArticleType);
+
+    if (categoryId && !supportedCategoryIds.includes(categoryId)) {
+        throw new Error(`"${categoryIdParameter}" is not a supported category ID. Please choose between the following: ${supportedCategoryIds.join(', ')}.`);
+    }
+
+    const responseData = await ofetch(`https://hw-media-cdn-mingchao.kurogame.com/akiwebsite/website2.0/json/G152/${language}/MainMenu.json`);
+
+    /** Category IDs mapped to their name. */
+    const categories = new Map(responseData.articleType.map(({ contentId, contentLabel }) => [contentId, contentLabel]) as [number, string][]);
+
+    // Ignore duplicate articles
+    const handledIds: number[] = [];
+
+    const itemPromises: Promise<DataItem>[] = [];
+
+    for (const article of responseData.article) {
+        // Limit the amount of items to display in the feed
+        if (handledIds.length >= limit) {
+            continue;
+        }
+
+        // Ignore duplicate articles
+        if (handledIds.includes(article.articleId)) {
+            continue;
+        }
+
+        // Filter on category
+        if (categoryId && article.articleType !== categoryId) {
+            continue;
+        }
+
+        handledIds.push(article.articleId);
+
+        const promise = async () => {
+            const description = await getArticleContent(language, article.articleId);
+            const category = categories.get(article.articleType);
+
+            if (description === null) {
+                logger.error(`Failed to fetch Wuthering Waves article content for article with ID ${article.articleId}`);
+            }
+
+            if (typeof description !== 'string') {
+                logger.error(`Expected Wuthering Waves article content for article with ID ${article.articleId} to be a string, but got ${typeof description} instead`);
+            }
+
+            return {
+                title: article.articleTitle,
+                description,
+                category: category ? [category] : undefined,
+                pubDate: timezone(parseDate(article.createTime), 0),
+                link: `https://wutheringwaves.kurogames.com/${language}/main/news/detail/${article.articleId}`,
+                guid: article.articleId,
+            } as DataItem;
+        };
+
+        itemPromises.push(promise());
+    }
```

Comment:
Use .map() instead of [for + push](https://www.measurethat.net/Benchmarks/Show/11532/0/push-vs-map).